scripts/generate_report.sh

```
@@ -8,8 +8,52 @@ log() {
+# date関数: OS依存の日付操作を抽象化
+get_date() {
+    if [[ "$IS_MACOS" == true ]]; then
+        case "$1" in
+            "24h_ago")
+                date -v-24H -u +"%Y-%m-%dT%H:%M:%SZ"
+                ;;
+            "month_start")
+                date -v1d -v0H -v0M -v0S -u +"%Y-%m-%dT%H:%M:%SZ"
+                ;;
+            "days_in_month")
+                days=$(cal $(date +"%m %Y") | awk 'NF {DAYS = $NF}; END {print DAYS}')
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)